### PR TITLE
Solution for issue 1098 - prompted passwords are not properly persisted into sessions

### PR DESCRIPTION
**Pull request by sebastianczech**
Pull request is proposition of solution for the issue #1098 , where prompted passwords are not properly persisted into sessions. Below is an example how it works after applying solution: 

[asciicast](https://asciinema.org/a/twmDRfaXgTS3nRNDH1INsq0pp)

This is my first contribution into open source, so all feedback is more than welcome ;)

### CODE REVIEW

File: httpie/cli/argtypes.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ class AuthCredentials(KeyValueArg):
     """Represents parsed credentials."""
 
     def has_password(self) -> bool:
-        return self.value is not None
+        return self.value
```
Comment (isidentical) [inline httpie/cli/argtypes.py:137]: I don't understand this change, why are we returning the password in `has_password`. `value` can be either a string (which is not a bool, check the function's signature) or `None`.

**Review by isidentical — CHANGES_REQUESTED** [review]
The solution seems a bit localized, see my comments. Other than that; I am not really a big fan of storing raw passwords locally, so I would personally go with re-asking it rather than storing it as plain text. CC: @jakubroztocil on the product decision

File: httpie/cli/argtypes.py (hunk 1 above)
Comment (sebastianczech) [reply httpie/cli/argtypes.py]: My mistake, I have fixed by changing it into ``bool(self.value)``. In some cases I saw, that password is ``None``, but sometimes it's empty string ``''``.

File: httpie/plugins/builtin.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ class BasicAuthPlugin(BuiltinAuthPlugin):
 
     # noinspection PyMethodOverriding
     def get_auth(self, username: str, password: str) -> HTTPBasicAuth:
+        self.raw_auth = username + ":" + password
```
[the remark this answers is not in the document]
Comment (sebastianczech) [reply httpie/plugins/builtin.py]: You are right. I removed from authentication plugin and move it into ``argparser.py`` , function ``_process_auth``. Is this approach acceptable ?

**Review by Amirjokr — APPROVED** [review]

**Review by isidentical — APPROVED** [review]
Thanks a lot for the PR @sebastianczech. I've noticed that the change you have made on the `has_password` broke the use case of empty passwords, so I manually fixed it and pushed it. The rest LGTM. Amazing first contribution 🎉

**Comment by sebastianczech** [comment]
Thank you @isidentical for review, fixing `has_password` and merging changes

**Pull request merged**